In [1]:
import pandas as pd
import numpy as np
import math

# Read in the data
df = pd.read_csv('./Data/free_text_input/Results/GPT4_top_prediction_RE.csv')
# split columns into multiple columns
df['dataset'] = df['file_path']

In [2]:
# format float to 2 decimal places
w = df.shape[0]
x = df[['predict_correct_in_top_5']].values.sum()
y = df[['predict_made_in_top_10']].values.sum()
print('{y} out of {w} predictions made in top 5 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 5 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

x = df[['predict_correct_in_top_10']].values.sum()
y = df[['predict_made_in_top_10']].values.sum()
print('{y} out of {w} predictions made in top 10 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 10 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

x = df[['predict_correct_in_top_50']].values.sum()
y = df[['predict_made_in_top_50']].values.sum()
print('{y} out of {w} predictions made in top 50 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 50 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))


102 out of 125 predictions made in top 5 (81.6%)
0 out of 102 predictions correct in top 5 (0.0%)
102 out of 125 predictions made in top 10 (81.6%)
4 out of 102 predictions correct in top 10 (3.92%)
78 out of 125 predictions made in top 50 (62.4%)
5 out of 78 predictions correct in top 50 (6.41%)


In [52]:
# This table is generated using ChatGPT-3

| Dataset | Top 5        |              | Top 10       |              | Top 50      |              |
| ------- | ------------ | ------------ | ------------ | ------------ | ----------- | ------------ |
|         | Predictions  | Correct      | Predictions  | Correct      | Predictions | Correct      |
| PMID    | 102 (81.6%)  | 0 (0.0%)     | 102 (81.6%)  | 4 (3.92%)    | 78 (62.4%)  | 5 (6.41%)    |
